lib/iris/fileformats/netcdf.py

```diff
@@ -1831,6 +1848,54 @@ def _create_cf_data_variable(self, cube, dimension_names, local_keys=None,
             The newly created CF-netCDF data variable.
 
         """
+        if 'packing' in kwargs:
+            packing = kwargs.pop('packing')
+            if packing:
+                scale_factor = None
+                add_offset = None
+                fill_value = None
+                masked = isinstance(cube.data, np.ma.core.MaskedArray)
+                if isinstance(packing, dict):
+                    if 'dtype' not in packing:
+                        msg = ("The dtype attribute is required for packing. ")
+                        raise ValueError(msg)
+                    dtype = np.dtype(packing['dtype'])
+                    if 'scale_factor' in packing:
+                        scale_factor = packing['scale_factor']
+                    if 'add_offset' in packing:
+                        add_offset = packing['add_offset']
+                    if 'fill_value' in packing:
+                        fill_value = packing['fill_value']
+                else:
+                    dtype = np.dtype(packing)
+                    cmax = cube.data.max()
+                    cmin = cube.data.min()
+                    n = dtype.itemsize * 8
+                    if masked:
+                        scale_factor = (cmax - cmin)/(2**n-2)
+                    else:
+                        scale_factor = (cmax - cmin)/(2**n-1)
+                    if dtype.kind == 'u':
+                        add_offset = cmin
+                    elif dtype.kind == 'i':
+                        if masked:
+                            add_offset = (cmax + cmin)/2
+                        else:
+                            add_offset = cmin + 2**(n-1)*scale_factor
+                if masked and not fill_value:
```

Comment:
@jswanljung Careful ... are you sure you don't want `if masked and fill_value is not None` here ... just in case someone wanted a `fill_value=0` (why? ... just because they can I suppose)